pipeline/normalize.py

```diff
@@ -13,15 +13,29 @@
 
 def normalize_category_names(df, mapping):
     def map_category_name(c):
-        if c in mapping:
+        if isinstance(c, str) and c.lower() in mapping:
             COUNTERS[c] += 1
-            return mapping[c]
+            return mapping[c.lower()]
         return c
     for col in ('tax1', 'tax2', 'tax3'):
         df[col] = df[col].apply(map_category_name)
     return df
 
 
+def replace_www_to_http(df, col):
+'''
+Modify website links that start with www
+
+Arguments: df - dataframe
+           col - column name in dataframe
+Returns: modified dataframe with replaced "www"
+         at the beginning
+'''
+    rows = df[col].str[:3] == "www"
+    df.loc[rows, col] = df[col].str.replace('www.', 'https://')
```

Comment:
Yes, you have a point. The main reason why I added this method was to make other websites that start with 'www' to be able to redirect to an actual website. When we were clicking those website with "www" they would show us "Not Found Page" error. However, after changing it to https://, they started to redirect to an actual original website. If you have some other advice on optimizing/choosing the solution, I would love to hear it!

Reply:
Hmm, that sounds to me like it's a problem with certain website links, and thus should be fixed in the original data spreadsheets for the affected websites, because i don't think this rule is correct to apply to _all_ websites. However, if you've checked that this usually works and doesn't cause problems, i'm okay with it!